articles/quickstart/native/android/03-session-handling.md

```diff
@@ -47,21 +47,55 @@ WebAuthProvider.init(auth0)
                 .withScheme("demo")
                 .withAudience(String.format("https://%s/userinfo", getString(R.string.com_auth0_domain)))
                 .withScope("openid offline_access")
-                .start(LoginActivity.this, callback);
+                .start(LoginActivity.this, webCallback);
 ```
 
-## Save the User's Credentials
+## Check for Tokens when the Application Starts
+
+::: panel Learn about refresh tokens
+Before you go further with this tutorial, read the [refresh token documentation](/refresh-token).
+It is important that you remember the following:
+* Refresh tokens must be securely saved.
+* Even though refresh tokens cannot expire, they can be revoked.
+* New tokens will never have a different scope than the scope you requested during the first login.
+:::
+
+You can simplify the way you handle user sessions using a Credential Manager class, which knows how to securely store, retrieve and renew credentials obtained from Auth0. Two classes are provided in the SDK to help you achieve this. Further read on how they work and their implementation differences is available in the [Saving and Renewing Tokens](/libraries/auth0-android/save-and-refresh-tokens.md) article. For this series of tutorials we're going to use the `SecureCredentialsManager` class as it encrypts the credentials before storing them in a private SharedPreferences file.
+
+
+Create a new instance of the Credentials Manager. When you run the application you'd first check if there are any previous saved credentials, in order to skip the login screen:
 
-Save the user's credentials obtained in the login success response.
+```java
+// app/src/main/java/com/auth0/samples/LoginActivity.java
+  Auth0 auth0 = new Auth0(this);
+  auth0.setOIDCConformant(true);
+  SecureCredentialsManager credentialsManager = new SecureCredentialsManager(this, new AuthenticationAPIClient(auth0), new SharedPreferencesStorage(this));
+
+  // Check if the activity was launched after a logout
+  if (getIntent().getBooleanExtra(KEY_CLEAR_CREDENTIALS, false)) {
+      credentialsManager.clearCredentials();
+  }
+
+  if (!credentialsManager.hasValidCredentials()) {
+    // Prompt Login screen.
+  } else {
+    // Obtain credentials and move to the next activity
+  }
+```
 
-::: warning
-Make sure you use a secure method.
+::: note
+Ideally a single class should have the knowledge of handling credentials, but you can share this instance across activities or create a new one every time is required as long as the Storage strategy persists the data in the same location. Check the `LoginActivity` class to understand how to achieve this in a single class.
 :::
 
+
+## Save the User's Credentials
+
+After a successful login response save the user's credentials on the Credentials Manager. Use the `saveCredentials` method.
```

Comment:
~After a successful login response save the user's credentials on the Credentials Manager. Use the `saveCredentials` method.~
After a successful login response, you can save the user's credentials using the `saveCredentials` method.

I prefer the word store instead of save.  However, as the method is called `saveCredentials` it is okay.